src/db.rs

```diff
@@ -71,6 +71,77 @@ impl DB {
         Ok(())
     }
 
+    pub fn migrate<P: AsRef<Path>>(&mut self, path: P) -> Result<()> {
+        if !self.dirs.is_empty() {
+            // FIXME: just for testing
+            #[cfg(not(debug_assertions))]
+            return Err(anyhow!(
+                "To prevent conflicts, you can only migrate from z with an empty \
+                 zoxide database!"
+            ));
+        }
+
+        let z_db_file = File::open(path).with_context(|| anyhow!("could not open z db file"))?;
+        let reader = BufReader::new(z_db_file);
+
+        for (idx, read_line) in reader.lines().enumerate() {
+            let line_number = idx + 1;
+            let line = if let Ok(line) = read_line {
+                line
+            } else {
+                eprintln!("could not read line {}: {:?}", line_number, read_line);
+                continue;
+            };
+
+            let split_line = line.rsplitn(3, '|').collect::<Vec<&str>>();
+
+            match split_line.as_slice() {
+                [_, rank_str, path_str] => {
+                    let path = PathBuf::from(path_str);
+                    let rank = rank_str
+                        .parse::<f64>()
+                        .with_context(|| anyhow!("could not parse rank"))?;
+
+                    // otherwise, the rank will get scaled down, depending on
+                    // how old the entry is (import in-place and let zoxide
+                    // scale down organically)
+                    let epoch = util::get_current_time()?;
+
+                    // TODO: does this *need* to be canonicalized?
```

Comment:
@ajeetdsouza What is your opinion on this?

The `z` database already stores `canonicalize`d paths, so I think it would make sense to remove `path_abs` and `path_str` below. As it stands, I have a little bit of duplicated code: `path_str` is transformed into a path (`PathBuf::from(path_str)`), which is canonicalized, which is then transformed into a string again (`let path_str = path_abs.to_str().....`).

Reply:
I decided to rely on the fact that the paths inside `z`'s database are already canonicalized. Let me know if you disagree with this course of action and I'd be happy to talk about it.

Reply:
I don't think we should make assumptions about the kind of data we're getting, especially since we're importing a format that a lot of different applications use (`z`, `z.lua`, `zsh-z`, etc.). While a rogue entry wouldn't really do much harm, an unresolved symbolic link would remain in the database for a while before its rank finally drops to < 1.

Reply:
You're right -- the smaller the amount of assumptions, the less of a chance of things going wrong.